src/node_api.cc

```diff
@@ -706,6 +706,40 @@ bool FindWrapper(v8::Local<v8::Object> obj,
   return true;
 }
 
+static void DeleteEnv(const v8::WeakCallbackInfo<napi_env__>& data) {
+  delete data.GetParameter();
+}
+
+napi_env get_env(v8::Local<v8::Context> context) {
+  napi_env result;
+
+  auto isolate = context->GetIsolate();
+  auto global = context->Global();
+
+  auto maybe_key = v8::String::NewFromUtf8(isolate, "N-API Environment",
+      v8::NewStringType::kInternalized);
+  CHECK(!maybe_key.IsEmpty());
+  auto key = v8::Private::ForApi(isolate, maybe_key.ToLocalChecked());
```

Comment:
Actually, storing the `napi_env` instead of the v8::Private means I lose the ability to pass a `napi_env` created internally to node-addon-api, so I wouldn't revert to the previous approach. Still, I would only introduce a mutex if caching it really is worth it for the load-module use case.